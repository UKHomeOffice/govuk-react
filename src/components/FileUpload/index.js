// https://github.com/alphagov/govuk-frontend/tree/master/src/components/file-upload

import React from "react";
import PropTypes from "prop-types";
import glamorous from "glamorous";

import Label from "../Label/index";
import LabelText from "../LabelText/index";
import ErrorText from "../ErrorText/index";
import HintText from "../HintText/index";

import { FONT_SIZE, NTA_LIGHT, MEDIA_QUERIES } from "../../constants/index";

const Input = glamorous.input({
  boxSizing: "border-box",
  fontFamily: NTA_LIGHT,
  WebkitFontSmoothing: "antialiased",
  MozOsxFontSmoothing: "grayscale",
  fontWeight: 400,
  textTransform: "none",
  fontSize: FONT_SIZE.SIZE_14,
  width: "100%",
  padding: "5px 4px 4px",
  [MEDIA_QUERIES.LARGESCREEN]: {
    width: "50%"
  },
  "[disabled]": {
    cursor: "auto"
  }
});

const FileUpload = ({ meta, children, hint, acceptedFormats }) => (
  <Label error={meta.error}>
    <LabelText error={meta.error}>{children}</LabelText>
    {hint && <HintText>{hint}</HintText>}
    {meta.touched && meta.error && <ErrorText>{meta.error}</ErrorText>}
    <Input type="file" accept={acceptedFormats} error={meta.error} />
  </Label>
);

FileUpload.defaultProps = {
  hint: null,
  meta: {},
  acceptedFormats: null
};

FileUpload.propTypes = {
  hint: PropTypes.string,
  meta: PropTypes.shape({
    active: PropTypes.bool,
    dirty: PropTypes.bool,
    dirtySinceLastSubmit: PropTypes.bool,
    error: PropTypes.any,
    initial: PropTypes.bool,
    invalid: PropTypes.bool,
    pristine: PropTypes.bool,
    submitError: PropTypes.any,
    submitFailed: PropTypes.bool,
    submitSucceeded: PropTypes.bool,
    touched: PropTypes.bool,
    valid: PropTypes.bool,
    visited: PropTypes.bool
  }),
  children: PropTypes.node.isRequired,
  acceptedFormats: PropTypes.string
};

export default FileUpload;
