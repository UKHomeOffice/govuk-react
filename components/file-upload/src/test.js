import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { createSerializer } from 'jest-emotion';
import * as emotion from 'emotion';

import FileUpload from './';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('FileUpload', () => {
  const example = 'example';
  const meta = {
    touched: true,
    error: example,
  };
  const wrapper = <FileUpload>{example}</FileUpload>;
  const wrapperError = (
    <FileUpload hint={example} meta={meta}>
      {example}
    </FileUpload>
  );
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
    ReactDOM.render(wrapperError, div);
  });

  it('matches snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('enzyme.mount');
  });
});
