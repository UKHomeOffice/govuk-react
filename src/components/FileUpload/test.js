import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, render } from "enzyme";
import FileUpload from "./";

describe("FileUpload", () => {
  const example = "example";
  const wrapper = <FileUpload>{example}</FileUpload>;

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(wrapper, div);
  });

  it("matches snapshot", () => {
    expect(shallow(wrapper)).toMatchSnapshot(`enzyme.shallow`);
    expect(mount(wrapper)).toMatchSnapshot(`enzyme.mount`);
    expect(render(wrapper)).toMatchSnapshot(`enzyme.render`);
  });
});
