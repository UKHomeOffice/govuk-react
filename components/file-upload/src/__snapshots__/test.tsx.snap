// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload matches snapshot: enzyme.mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  margin-bottom: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c2 {
  box-sizing: border-box;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  text-transform: none;
  font-size: 14px;
  line-height: 1.1428571429;
  width: 100%;
  padding: 5px 4px 4px;
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 0;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 16px;
    line-height: 1.25;
    width: 50%;
  }
}

<FileUpload
  meta={Object {}}
>
  <Label
    error={false}
  >
    <src__StyledLabel
      error={false}
    >
      <label
        className="c0"
      >
        <LabelText>
          <src__StyledLabelText>
            <span
              className="c1"
            >
              example
            </span>
          </src__StyledLabelText>
        </LabelText>
        <src__StyledInput
          type="file"
        >
          <input
            className="c2"
            type="file"
          />
        </src__StyledInput>
      </label>
    </src__StyledLabel>
  </Label>
</FileUpload>
`;
